backend/game_engine/matchmaker.py

```diff
@@ -126,3 +134,116 @@ def generate_join_game_verification_link(self, game_id: Text, player_id: Text) -
 
     def generate_game_info_link(self, game_id: Text) -> Text:
         return "{}/game-info/{}".format(_VIR_US_FE_HOSTNAME, game_id)
+
+class MatchMakerInterface(ABC):
+
+    @classmethod
+    @abstractmethod
+    def generate_teams(cls, game_id: Text, players: list, team_size: int) -> dict:
+        # Take in game_id and list of Players and return a list of Teams
+        return
+
+    @classmethod
+    @abstractmethod
+    def generate_teams_tribes(cls, game_id: Text, players: list, game_options: GameOptions) -> dict:
+        # Take in game_id, list of Players, and gameOptions, and return a list of Teams and Tribes
+        return
+
+class MatchMakerRoundRobin(MatchMakerInterface):
+    @classmethod
+    def generate_teams(cls, game_id: Text, players: list, team_size: int) -> dict:
+        teams = []
+        count_players = len(players)
+        if count_players<team_size:
+            raise Exception("Insufficient players for given team size")
+        for n in range(0, math.floor(count_players / team_size)):
+            team = database.Team(
+                id=str(uuid.uuid4()),
+                name="team/{}".format(n),
+            )
+            teams.append(team)
+
+        count_teams = len(teams)
+        for n, player in enumerate(players):
+            team = teams[n % count_teams]
+            player.team_id = team.id
+            team.size += 1
+
+        teams_dict = {}
+        for team in teams:
+            teams_dict[team.id] = team.to_dict()
+            del teams_dict[team.id]['id']
+        return teams_dict
+
+    @classmethod
+    def generate_teams_tribes(cls, game_id: Text, players: list, game_options: GameOptions, gamedb: Database) -> dict:
+        tribes = []
+        teams = []
+        count_players = len(players)
+        if count_players < game_options.target_team_size:
+            raise Exception("Insufficient players for given team size")
```

Comment:
update 'Exception' to a subclass scoped within this module, for example 'MatchMakerException'. 

What happens when this gets raised?

Reply:
For consistency with GameError I named this 'MatchMakerError'. I don't have a way of handling this exception right now so when this gets raised the game won't start, and the players won't be notified of this. The game will also be marked as 'started', so it will basically be stuck in limbo. However because this is running on a separate process, at least the whole region server won't crash. 

In theory this shouldn't happen as long as we set the 'min_players' value correctly when starting the server, but I think this is definitely an area that could be improved.